src/server/use-request.js

```diff
@@ -0,0 +1,30 @@
+const { AsyncLocalStorage } = require('node:async_hooks')
+
+const requestStorage = new AsyncLocalStorage()
+
+/**
+ * An express middleware. Captures the request in AsyncLocalStorage for later
+ * retrieval.
+ */
+function requestProviderMiddleware (req, res, next) {
+  requestStorage.run(req, next)
```

Comment:
The documentation seems to imply that garbage collection can only happen once this (experimental?!) api is called?
https://nodejs.org/api/async_context.html#asynclocalstoragedisable

If that's true, then I think we should be mindful of the amount of extra data we're adding into the store, and a smaller object would be safer here instead of using the whole req

Reply:
Oh, weird!  Good to call this out, I had not spotted it.

I just re-read the spec and I think this is only talking about the memory reserved for the asynclocalstorage container itself.  It specifically states that the "store" content can be garbage collected safely — so I don't think we have to worry too much about the garbage collection of the request object itself.

> Calling asyncLocalStorage.disable() is required before the asyncLocalStorage can be garbage collected. This does not apply to stores provided by the asyncLocalStorage, as those objects are garbage collected along with the corresponding async resources.